src/components/ConfirmationModal/ConfirmationModal.tsx

```diff
@@ -0,0 +1,133 @@
+import React from "react";
+import { Modal, StyledModalContent, StyledModalFeet } from "../Modal";
+import Box from "../Box";
+import Typography from "../Typography";
+import Button from "../Button";
+import { ButtonVariant } from "../Button/Button";
+import BackgroundIcon from "../BackgroundIcon";
+
+export interface ConfirmationModalButton {
+  variant: ButtonVariant;
+  text: string;
+  onClick: () => void;
+  isLoading?: boolean;
+  disabled?: boolean;
+}
+
+export interface ConfirmationModalProps {
+  /**
+   * Whether the modal is open or not.
+   */
+  isOpen: boolean;
+
+  /**
+   * Callback function to close the modal.
+   */
+  onClose: () => void;
+
+  /**
+   * Title of the ConfirmationModal.
+   */
+  title: string;
+
+  /**
+   * Text content of the ConfirmationModal.
+   */
+  text: string;
+
+  /**
+   * Background icon for the ConfirmationModal.
+   */
+  backgroundIcon: string;
+
+  /**
+   * Array of button configurations for the ConfirmationModal.
+   */
+  buttons: ConfirmationModalButton[];
+}
+
+/**
+ * `ConfirmationModal` is a reusable, customizable modal component for displaying
+ * confirmation prompts to users. It features a background icon, text content, and
+ * configurable buttons for handling user actions.
+ *
+ * Component Demo: [ConfirmationModal](https://ui.govconnex.com/?path=/story/components-ConfirmationModal--example)
+ *
+ */
+
+const ConfirmationModal = ({
+  isOpen,
+  onClose,
+  title,
+  text,
+  backgroundIcon,
+  buttons,
+}: ConfirmationModalProps) => (
+  <Modal
+    open={isOpen}
+    onClose={onClose}
+    enableClickAway
+    disablePortal
+    width="400px"
+  >
+    <StyledModalContent>
+      <Box
+        cs={{
+          display: "flex",
+          alignItems: "flex-start",
+          gap: "spacing.sm",
+          marginTop: "spacing.sm",
+          marginBottom: "spacing.sm",
+        }}
+      >
+        <Box>
+          <BackgroundIcon icon={backgroundIcon} size="2x" />
+        </Box>
+        <Box
+          cs={{
+            display: "flex",
+            flexDirection: "column",
+            gap: "spacing.xs",
+            flexGrow: 1,
+          }}
+        >
+          <Typography variant="heading" size="xs" noMargin>
+            {title}
+          </Typography>
+          <Typography
+            variant="body"
+            noMargin
+            color="core.content.contentSecondary"
+          >
+            {text}
+          </Typography>
+        </Box>
+      </Box>
+    </StyledModalContent>
+    <StyledModalFeet>
+      <Box
```

Comment:
q; would it support a button group something like this?  
https://www.figma.com/file/pQqcx1z6rSV8MVmokydQMr/%F0%9F%93%90-Bipartisan-Design-System?node-id=812-4280&t=oJ692gMr22MTLCEa-4

where one btn is way left and the rest are on right, or maybe vise versa

Reply:
Yes, eventually. Out of scope for this PR though, this is just the base functionality.